opm/grid/common/SubGridView.hpp

```diff
@@ -0,0 +1,371 @@
+        template <PartitionIteratorType pit>
+        struct Partition {
+            /** \brief iterator over a given codim and partition type */
+            using Iterator = SubIterator;
+        };
+
+    };
+
+    enum { conforming = Traits::conforming };
+    enum { dimension = GridImp::dimension };
+
+public:
+
+    /// Construct a view of the codim 0 entities that can be constructed from the seeds input.
+    ///
+    /// The seeds input is moved from and will be in a valid but indeterminate state after the call.
+    SubGridView(const Grid& grid,
+                std::vector<typename Codim<0>::Entity::EntitySeed>&& seeds,
+                const bool overlap = true)
+        : grid_(&grid)
+        , subset_(std::move(seeds))
+        , num_owned_(subset_.size())
+    {
+        // Nothing more to do if we do not want to have overlap entities.
+        if (!overlap) {
+            return;
+        }
+
+        // Add neighbouring not-owned entities to subset_
+        using Seed = typename Codim<0>::Entity::EntitySeed;
+        std::unordered_set<int> owned;
+        std::unordered_map<int, Seed> neighbors;
+        const auto& iset = grid_->leafIndexSet();
+        const auto& leaf_view = grid_->leafGridView();
+        for (const auto& seed : subset_) {
+            // Add this entity to the set of owned indices.
+            const auto& entity = grid_->entity(seed);
+            owned.insert(iset.index(entity));
+            // Iterating over all intersections, ...
+            const auto end = leaf_view.iend(entity);
+            for (auto it = leaf_view.ibegin(entity); it != end; ++it) {
+                if (it->boundary()) {
+                    continue;
+                }
+                if (it->neighbor()) {
+                    const auto outside_entity = it->outside();
+                    // ...for all neighbour entities, add to neighbors.
+                    neighbors.try_emplace(iset.index(outside_entity), outside_entity.seed());
+                }
+            }
+        }
+        // Now that owned is complete, we can eliminate any owned entries.
+        std::map<int, Seed> unowned_neighbors;
+        for (const auto& nb : neighbors) {
+            if (owned.count(nb.first) == 0) {
+                unowned_neighbors.insert(nb);
+            }
+        }
+        subset_.resize(subset_.size() + unowned_neighbors.size());
+        std::size_t count = num_owned_;
+        for (const auto& [index, seed] : unowned_neighbors) {
+            subset_[count] = seed;
+            ++count;
+        }
+        assert(count == subset_.size());
+    }
+
+    /** \brief obtain a const reference to the underlying hierarchic grid */
+    const Grid& grid() const
+    {
+        assert(grid_);
+        return *grid_;
+    }
+
+    /** \brief obtain the index set */
+    // const IndexSet& indexSet() const // Not implemented
+
+    /** \brief obtain number of entities in a given codimension */
+    int size(int codim) const
```

Comment:
Only correct for codim 0, but even CpGrid supports this for vertices

Reply:
See previous answer.